### DISCUSSION TITLE
TUI for building waybar

### DISCUSSION THREAD

**Discussion by GeorgeAndreiBals** [prasanthrangan/hyprdots#343] [Ideas]
Maybe I'm not the only one but some of the waybar presets are buggy so what I've thought it may be cool is a little script that would let you choose the modules from waybar you want to use, waybar's placement (top/bottom) etc.
I might dab into this when I have some more time, but I though about writing this here so I don't forget about it 😄

**Comment by kRHYME7** [comment]
Is this not the same with the ``` ~/.config/waybar/config.ctl ``` ? , if so then that would be great to have a more convenient feature.

**Reply by prasanthrangan** [reply]
Yup, `~/.config/waybar/config.ctl` already kinda does that...
So if we gonna do this I guess will it be better to build a GUI that shows all the available modules that we can drag and drop to configure waybar!

**Reply by T-Crypt** [reply]
I've actually been working on this! Its a work in progress

![231019_12h38m00s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/132922589/573f5b46-cc87-41d5-bc62-14901294fe06)

**Reply by Balssh** [reply]
Yup, something like this is what I had in mind when writing this post, but I didn't know to explain myself better :smile:

**Reply by prasanthrangan** [reply]
Amazing... is this gtk-layer-shell window? I'll design a concept GUI for the interface and share it here... We'll see which toolkit to use to achieve it!

**Comment by T-Crypt** [comment]
> Maybe I'm not the only one but some of the waybar presets are buggy so what I've thought it may be cool is a little script that would let you choose the modules from waybar you want to use, waybar's placement (top/bottom) etc.
> 
> I might dab into this when I have some more time, but I though about writing this here so I don't forget about it 😄 

config.ctl in the waybar directory already does this. 

You can just remove the modules you don't want from the config.ctl to customize your waybar. 

Make sure to leave the spacing as it is.
